Revamp demo page and add EE demo (#9430)
marmelab/react-admin: docs/Demos.md

If you want to see what react-admin is capable of, or if you want to learn from apps built by seasoned react-admin developers, check out these demos.

## Overview

Discover our demos that show you the react-admin potential and help you familiarize with it.

<style>
    .demos-list {
        display: grid;
        grid-template-columns: repeat(2, 1fr);

>         grid-template-rows: repeat(3, 1fr);

Reviewer: I'm not a css grid specialist, but I'm a bit surprised you have to set both `grid-template-columns` and `grid-template-rows`. To me, you could keep only `grid-template-columns`.

        grid-gap: 10px;
    }

    @media (max-width: 600px) {
        .demos-list {
            display: flex;
            flex-wrap: wrap;
            gap: 10px;
        }

>     }

Reviewer: Why switch to `flex` on mobile mode? Why not keep `grid` and only change `grid-template-columns` to be `repeat(1, 1fr);`?

    .card {

>         box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);

Reviewer 2: The shadow is a bit strong in my opinion
```suggestion
        box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
```

        transition: 0.3s;
        background-color: rgba(243, 246, 249, 0.4);
        border: 1px solid rgba(0,0,0,0.25);

>         border-radius: 30px;

Reviewer: `border-radius` is a bit too high IMHO. It doesn't fit well with the rest of the UI, and crops out part of the videos in a weird way, especially on mobile.

![image](https://github.com/marmelab/react-admin/assets/14542336/b2639142-4d5c-4e61-9610-e8740006919f)

But this is a taste issue, you may want to gather more feedback from the team before changing anything.

        margin: 0;
        display: flex;
        flex-direction: column;
    }

    .card:hover {
        box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
    }

    .card > img, .card > video {
        border-radius: 15px;

>         width: 100%;

Reviewer 2: Videos are not centered, there is more right margin because of the `.markdown-section img, .markdown-section video` rules.
![image](https://github.com/marmelab/react-admin/assets/7949510/7fa04efb-1a67-4d62-875c-0a9de2121da2)
This code will fix this:
```suggestion
        max-width: 100%;
```

>     }

Reviewer: It's a bit of a shame that the card's video/img/iframe do not all have the same height, as it creates inconsistencies in the layout...

![image](https://github.com/marmelab/react-admin/assets/14542336/e4ee43e6-ba4d-4a69-8944-f034f560fb35)

Ideally they should all be the same height, but that would involve re-making some of the videos, which can be time consuming...

Alternatively, maybe it is enough to only reserve the same height for each item?

    }

    #player {
        width: 100%;
    }

    .content-card {
        padding: 2px;
        margin-left: 1rem;
        margin-right: 1rem;
        text-decoration: none !important;
    }

    .title-card {
        margin-bottom: 0.4rem !important;
        font-size: 1.75em;
        color: black;
    }

    .description-card {
        color: rgb(100,100,100);
    }

    .card-footer {
        flex-grow: 1;
        display: flex;
        flex-direction: column-reverse;
    }

    .links-container {
        padding: 2px;
        margin-left: calc(1rem - 10px);
        margin-right: calc(1rem - 10px);
        display: flex;
        justify-content: space-between;
    }

    .link {
        color: rgb(0, 127, 255) !important;
        padding: 10px;
        border-radius: 15px
    }

    .link:hover {
        background-color: rgba(0, 127, 255, 0.07) !important;
    }

    .no-decoration, .link {
        text-decoration: none !important;
    }
</style>

<div class="demos-list">
    <div class="card">
        <iframe src="https://player.vimeo.com/video/474999017?autoplay=1&loop=1&muted=1" frameborder="0" allowfullsreen></iframe>
        <div class="content-card">

>             <a href="#e-commerce" class="no-decoration">

Reviewer 2: I'm a bit lost when I click the links anchor. I would have imagined dedicated pages for each sections instead, that explain further what they demonstrate. 
What do you think about it?

Author: I'm a bit worried about the number of pages this will create. won't it overload the navbar?

Reviewer: For the record, I disagree with @adguernier , IMO the anchors are fine and we should keep them that way.
To be discussed with @fzaninotto then.

                <p class="title-card"><b>E-commerce</b></p>
                <p  class="description-card">The admin of a fictional poster shop, allowing to manage sales, products, customers and reviews. Built by the core team.</p>
            </a>
        </div>
        <div class="card-footer">
            <div class="links-container">
                <p><b><a href="https://marmelab.com/react-admin-demo/" class="demo link">Demo</a></b></p>
                <p><b><a href="https://github.com/marmelab/react-admin/tree/master/examples/demo" class="source-code link">Source code</a></b></p>
            </div>
        </div>
    </div>
    <div class="card">
        <video src="https://github.com/marmelab/react-admin/assets/131013150/93ec41f5-81bf-45e1-940b-d5aeccfe628f" autoplay loop></video>
        <div class="content-card">
            <a href="#e-commerce-enterprise-version" class="no-decoration">

>                 <p class="title-card"><b>E-commerce enterprise version</b></p>

Reviewer: ```suggestion
                <p class="title-card"><b>E-commerce Enterprise</b></p>
```

>                 <p  class="description-card">Based on the E-commerce demo and upgraded with some ReactAdmin-Enterprise modules.</p>

Reviewer: ```suggestion
                <p  class="description-card">Based on the E-commerce demo and upgraded with some React Admin Enterprise features. Built by the core team.</p>
```

- [Custom page with static list](https://github.com/marmelab/react-admin/blob/master/examples/demo/src/routes.tsx)
- [Edit view in a sidebar](https://github.com/marmelab/react-admin/blob/master/examples/demo/src/reviews/ReviewList.tsx)

## E-commerce enterprise version

> Based on the E-commerce demo and upgraded with some ReactAdmin-Enterprise modules.

Reviewer: ```suggestion
Based on the E-commerce demo and upgraded with some React Admin Enterprise features.
```

> <video src="https://github.com/marmelab/react-admin/assets/131013150/93ec41f5-81bf-45e1-940b-d5aeccfe628f" controls="controls" style="max-width: 100%"></video>

Reviewer: Why not use the same tags (and style) as other videos in the docs?

In particular, it seems the video player is a but cropped to the right.

![image](https://github.com/marmelab/react-admin/assets/14542336/ab974bf7-bd54-4996-85af-b5bab552bbfe)

Also, can you not include the browser frame in the video?

Lastly, apart from a few exceptions, it seems to me you are not focusing enough on the Enterprise features included in this demo, and you are instead demonstrating mainly the same features as the OSS version of this demo. We can have a look together of the Enterprise features we should highlight if you need.

- [AccordionForm](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/products/ProductEdit.tsx)
- [Breadcrumb](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/layout/Breadcrumb.tsx)
- [SolarLayout](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/layout/Layout.tsx)
- [Locks](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/products/GridList.tsx)
- [ListLive](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/orders/OrderList.tsx)
- [RealtimeDataProvider](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/dataProvider/addRealtimeMethodsWithFakeTransport.ts)
- [CompleteCalendar](.https://github.com/marmelab/ra-enterprise-demo/blob/master/src/visits/VisitList.tsx)
- [Audit Log](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/i18n/en.ts)
- [SearchWithResult](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/layout/SearchSubMenu.tsx)
- [TreeWithDetails](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/categories/CategoryList.tsx)
- [MarkdownField](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/products/ProductPreview.tsx)

> - [MarkdownInput](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/products/ProductCreate.tsx)

Reviewer: This list is missing:

- [Tours](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/tours/TourList.tsx)
- [EditableDatagrid](https://github.com/marmelab/ra-enterprise-demo/blob/master/src/stores/StoreList.tsx)
